diff --git a/node_modules/expo-audio/ios/AudioRecords.swift b/node_modules/expo-audio/ios/AudioRecords.swift
index 33db27c..757e756 100644
--- a/node_modules/expo-audio/ios/AudioRecords.swift
+++ b/node_modules/expo-audio/ios/AudioRecords.swift
@@ -64,6 +64,7 @@ struct Metadata: Record {
   @Field var artist: String?
   @Field var albumTitle: String?
   @Field var artworkUrl: URL?
+  @Field var durationSeconds: Double?
 }
 
 struct LockScreenOptions: Record {
diff --git a/node_modules/expo-audio/ios/MediaController.swift b/node_modules/expo-audio/ios/MediaController.swift
index e977f57..22f52bb 100644
--- a/node_modules/expo-audio/ios/MediaController.swift
+++ b/node_modules/expo-audio/ios/MediaController.swift
@@ -80,7 +80,11 @@ class MediaController {
       info.removeValue(forKey: MPMediaItemPropertyPlaybackDuration)
       info.removeValue(forKey: MPNowPlayingInfoPropertyElapsedPlaybackTime)
     } else {
-      info[MPMediaItemPropertyPlaybackDuration] = player.duration
+      if let userDuration = player.metadata?.durationSeconds {
+        info[MPMediaItemPropertyPlaybackDuration] = userDuration
+      } else {
+        info[MPMediaItemPropertyPlaybackDuration] = player.duration
+      }
       info[MPNowPlayingInfoPropertyElapsedPlaybackTime] = player.currentTime
     }
     info[MPNowPlayingInfoPropertyPlaybackRate] = player.isPlaying ? player.ref.rate : 0.0
